repository_test.go

```diff
@@ -448,6 +448,24 @@ func (s *RepositorySuite) TestPlainCloneWithRecurseSubmodules(c *C) {
 	c.Assert(cfg.Submodules, HasLen, 2)
 }
 
+func (s *RepositorySuite) TestPlainCloneNoCheckout(c *C) {
```

Comment:
can we ensure that HEAD is pointing to the commit should be pointing?

Reply:
@mcuadros Take a look now - I've added a HEAD check